### PR TITLE
Fixes issue #4323, bugzilla automation long-run oscap testcase added.

### PR DESCRIPTION
**Pull request by san7ket**
Fixed a delete locator for oscap report.
Added a test case to cover https://bugzilla.redhat.com/show_bug.cgi?id=1420439

### CODE REVIEW

**Comment by san7ket** [comment]
 pytest tests/foreman/longrun/test_oscap.py
 test session starts 

platform linux2 -- Python 2.7.12, pytest-3.0.6, py-1.4.31, pluggy-0.4.0
rootdir: /home/sjagtap/PycharmProjects/robottelo, inifile:
plugins: cov-2.3.1, xdist-1.15.0
collected 8 items

2017-02-20 12:53:30 - conftest - DEBUG - Collected 8 test cases

tests/foreman/longrun/test_oscap.py ...s....

 0 tests deselected 
7 passed, 1 skipped in 1053.59 seconds

**Comment by kbidarkar** [comment]
ACK

**Comment by renzon** [comment]
Check the lines without coverage: https://coveralls.io/builds/10231171/source?filename=robottelo%2Fapi%2Futils.py

File: tests/foreman/longrun/test_oscap.py [hunk 1]
```diff
@@ -269,6 +178,125 @@ def test_positive_upload_to_satellite(self):
...
+                )
+                self.oscapreports.delete(host, really=True)
+                for _ in range(2):
+                    vm.run(u'puppet agent -t 2> /dev/null')
+                updated_result = vm.run(
+                    u'cat /etc/foreman_scap_client/config.yaml'
+                    '| grep profile'
+                )
+                self.assertIsNot(result, updated_result)
+                self.assertEqual(result.return_code, 0)
```
Comment (oshtaier) [inline tests/foreman/longrun/test_oscap.py]: `self.assertEqual(updated_result.return_code, 0)`?

File: tests/foreman/longrun/test_oscap.py [hunk 2]
```diff
@@ -269,6 +178,125 @@ def test_positive_upload_to_satellite(self):
...
+                    domain_name=vm._domain,
+                    parameters_list=[
+                        ['Host', 'Lifecycle Environment',
+                         self.config_env['env_name']],
+                        ['Host', 'Content View', self.config_env['cv_name']],
+                        ['Host', 'Host Group', vm_values.get('hgrp')],
+                        ['Host', 'Reset Puppet Environment', True],
+                    ],
+                )
+                session.nav.go_to_hosts()
```
Comment (oshtaier) [inline tests/foreman/longrun/test_oscap.py]: Redundant line or it was designed to verify something afterwards?

File: tests/foreman/longrun/test_oscap.py [hunk 3]
```diff
@@ -269,6 +178,125 @@ def test_positive_upload_to_satellite(self):
                     # Satellite6.
                     self.assertTrue(self.oscapreports.search(host))
 
+    @skip_if_bug_open('bugzilla', 1420439)
+    @tier4
+    def test_positive_push_updated_content(self):
+        """Perform end to end oscap test, and push
```
Comment (oshtaier) [inline tests/foreman/longrun/test_oscap.py]:
```
        """Perform end to end oscap test, and push the updated scap content
        after first run.
```

**Comment by oshtaier** [comment]
Please, fix the comments, do rebase and re-test (of course if all other reviewers finished to not run tests again)

File: tests/foreman/longrun/test_oscap.py [hunk 4]
```diff
@@ -269,6 +178,125 @@ def test_positive_upload_to_satellite(self):
...
+                    profile=OSCAP_PROFILE['firefox'],
+                    period=OSCAP_PERIOD['weekly'],
+                    period_value=OSCAP_WEEKDAY['friday'],
+                )
+                self.oscapreports.delete(host, really=True)
+                for _ in range(2):
+                    vm.run(u'puppet agent -t 2> /dev/null')
+                updated_result = vm.run(
+                    u'cat /etc/foreman_scap_client/config.yaml'
+                    '| grep profile'
```
Comment (kbidarkar) [inline tests/foreman/longrun/test_oscap.py]: As the test talks about updating oscap content, we need to check for "content" and not "profile". I know this should also work but, can you please check if we can instead grep for "content" and not "profile".

File: tests/foreman/longrun/test_oscap.py (hunk 2 above)
Comment (san7ket) [inline tests/foreman/longrun/test_oscap.py]: Cleaned it out.

File: tests/foreman/longrun/test_oscap.py (hunk 1 above)
Comment (san7ket) [inline tests/foreman/longrun/test_oscap.py]: corrected this.

File: tests/foreman/longrun/test_oscap.py (hunk 4 above)
Comment (san7ket) [inline tests/foreman/longrun/test_oscap.py]: Will grep for content_path instead of profile

**Comment by san7ket** [comment]
Conflicts resolved , rebase done , comments addressed 

collected 8 items
2017-03-07 14:09:40 - conftest - DEBUG - Found WONTFIX in decorated tests ['1269196', '1245334', '1217635', '1226425', '1156555', '1204686', '1267224', '1103157', '1230902', '1214312', '1079482']

2017-03-07 14:09:40 - conftest - DEBUG - Collected 8 test cases

tests/foreman/longrun/test_oscap.py s
== 7 tests deselected 1 skipped, 7 deselected in 250.24 seconds ====

**Comment by san7ket** [comment]
2017-03-07 15:29:28 - robottelo.ssh - INFO - Destroying Paramiko client 0x7fbc81c07550
2017-03-07 15:29:28 - robottelo.ssh - INFO - Destroyed Paramiko client 0x7fbc81c07550
2017-03-07 15:29:31 - robottelo.ssh - INFO - Instantiated Paramiko client 0x7fbc762df950
2017-03-07 15:29:31 - robottelo.ssh - DEBUG - Connected to [10.8.29.249]
2017-03-07 15:29:31 - robottelo.ssh - DEBUG - >>> cat /etc/foreman_scap_client/config.yaml| grep content_path
2017-03-07 15:29:31 - robottelo.ssh - DEBUG - <<< stdout
  :content_path: '/var/lib/openscap/content/654f841b9386f771d3999f855f28bfed01e2be4036774103e822950c1e4230aa.xml'

2017-03-07 15:29:31 - robottelo.ssh - INFO - Destroying Paramiko client 0x7fbc762df950
2017-03-07 15:29:31 - robottelo.ssh - INFO - Destroyed Paramiko client 0x7fbc762df950
2017-03-07 15:29:34 - robottelo.ssh - INFO - Instantiated Paramiko client 0x7fbc762e5610
2017-03-07 15:29:34 - robottelo.ssh - DEBUG - Connected to [10.8.29.249]
2017-03-07 15:29:34 - robottelo.ssh - DEBUG - >>> awk -F "/" '/download_path/ {print $4}' /etc/foreman_scap_client/config.yaml
2017-03-07 15:29:35 - robottelo.ssh - DEBUG - <<< stdout
1

2017-03-07 15:29:35 - robottelo.ssh - INFO - Destroying Paramiko client 0x7fbc762e5610
2017-03-07 15:29:35 - robottelo.ssh - INFO - Destroyed Paramiko client 0x7fbc762e5610
2017-03-07 15:29:37 - robottelo.ssh - INFO - Instantiated Paramiko client 0x7fbc762e52d0
2017-03-07 15:29:37 - robottelo.ssh - DEBUG - Connected to [10.8.29.249]
2017-03-07 15:29:37 - robottelo.ssh - DEBUG - >>> foreman_scap_client 1
2017-03-07 15:29:40 - robottelo.ssh - DEBUG - <<< stdout
File /var/lib/openscap/content/654f841b9386f771d3999f855f28bfed01e2be4036774103e822950c1e4230aa.xml is missing. Downloading it from proxy
Download scap content xml from: https://dell-per920-02.khw.lab.eng.bos.redhat.com:9090/compliance/policies/1/content
DEBUG: running: oscap xccdf eval --profile xccdf_org.ssgproject.content_profile_stig-firefox-upstream --results-arf /tmp/d20170307-11851-k9x3sh/results.xml /var/lib/openscap/content/654f841b9
386f771d3999f855f28bfed01e2be4036774103e822950c1e4230aa.xml
Scan failed
This content points out to the remote resources. Use `--fetch-remote-resources' option to download them.

2017-03-07 15:29:40 - robottelo.ssh - DEBUG - <<< stderr
WARNING: Skipping http://www.redhat.com/security/data/oval/Red_Hat_Enterprise_Linux_6.xml file which is referenced from XCCDF content
Profile "xccdf_org.ssgproject.content_profile_stig-firefox-upstream" was not found. Get available profiles using:
$ oscap info "/var/lib/openscap/content/654f841b9386f771d3999f855f28bfed01e2be4036774103e822950c1e4230aa.xml"


This will pass , once the bug is fixed

**Comment by kbidarkar** [comment]
ACK , after 2nd review.

May need fixes for coverage.

**Comment by oshtaier** [comment]
Ready to merge. Not sure what we can do here in terms of coverage

**Comment by san7ket** [comment]
@kbidarkar @oshtaier Should I create a issue to track it separately , and we can merge this

**Comment by san7ket** [comment]
Issue created https://github.com/SatelliteQE/robottelo/issues/4413

**Comment by kbidarkar** [comment]
Will merge this PR, as not everyone is familiar with handling coverage issues right away. We can work towards learning and handling coverage part and then raise PR for them soon.

**Pull request merged**